client/src/App.js

```diff
@@ -31,10 +32,24 @@ function App() {
   return (
     <Main>
       {data.map(data => (
-        <Card author={data.author} text={data.text} key={data.id}/>
+        <Card author={data.author} text={data.text} key={data.id} />
       ))}
+      <Footer onCreateQuestion={handleCreateQuestion} />
     </Main>
   )
+  function handleCreateQuestion({ text, author }) {
+    let i = data.length
+    i++
+    const newQuestion = [
+      ...data,
+      {
+        id: i,
+        text: text,
+        author: author,
+      },
+    ]
+    setData(newQuestion)
+  }
```

Comment:
this function won't work as expected once you reach user story 4. 
if you have for example 3 items in your data array and delete the second one you will have 2 left. the last one then has the id 3. if you then add a new question the id of the new question will be 3, too
hint: nanoid

Reply:
Thank you! We switched to nanoid! :-)